src/components/snackbar/use-snackbar-vertical-position-animation.ts

```diff
@@ -70,10 +73,27 @@ export const useSnackbarVerticalPositionAnimation = (
     viewHeightIncludingShadow,
   ]);
 
+  // when the Snackbar is disabled, and textContent changes to make the Snackbar taller,
+  // it takes a moment for it to move into the new hiddenY,
+  // however the Snackbar should not be visible during this transition.
+  // isHidden remains true during this transition.
+  const [isHidden, setIsHidden] = useState(!snackbarIsVisible);
```

Comment:
Is this state necessary? Can't it just be derived from `snackbarIsVisible` and `isInHiddenY` during render?

Reply:
Yes I think it's necessary. During the transition triggered by the height change while hidden, `snackbarIsVisible` is `false` and `isInHiddenY` is `false`. This happens to be exactly the same state as when it exits normally, in which case it should show..